PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/ServicePrincipalLoginDialog.java

```diff
@@ -256,33 +255,40 @@ public static String findTextInClipboard(Predicate<String> func) {
     }
 
     private void uiTextComponents2Json() {
-        if (intermediateState) {
+        if (!intermediateState.compareAndSet(false, true)) {
```

Comment:
as mentioned in daily scrum.
you can solve the problem in a more elegant way by comparing the data-model(but not the view model) of the upper inputs-form and lower json-form, and then sync to the other when one is changed.